### PR TITLE
Added HTC material for turbulent flow in annular ducts

### LINKED ISSUE #26389: HTC correlation using annular flow formulation from Gnielinski
## Reason
Wall heat transfer in annular flows requires specific treatment.

## Design
This would add a new wall heat transfer coefficient material implementing the annular Gnielinski correlation given in the following reference:

Volker Gnielinski (2009) Heat Transfer Coefficients for Turbulent Flow in Concentric Annular Ducts, Heat Transfer Engineering, 30:6, 431-436, DOI: 10.1080/01457630802528661

## Impact
New class. No impact to existing input files.

### PR DESCRIPTION
**Pull request by joshuahansel**
Closes #26389

### CODE REVIEW

File: modules/thermal_hydraulics/src/materials/ADWallHTCGnielinskiAnnularMaterial.C [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        mooseWarning("This material uses a correlation that is valid for Re > 1e4, but the "
+                     "material was evaluated with an Re = " +
+                     Moose::stringify(MetaPhysicL::raw_value(Re)) +
+                     ". This and all subsequent evaluations will be given a lower bound of 1e4."));
+    Re = 1e4;
+  }
+  const ADReal Re_star =
+      Re * ((1 + _a * _a) * std::log(_a) + (1 - _a * _a)) / (std::pow(1 - _a, 2) * std::log(_a));
+
+  const ADReal f_ann = std::pow(1.8 * std::log10(Re_star) - 1.5, -2.0);
```
Comment (licharlot) [inline modules/thermal_hydraulics/src/materials/ADWallHTCGnielinskiAnnularMaterial.C:121]: In the documentation page, the 1.8 factor in not to the -2.

File: modules/thermal_hydraulics/src/materials/ADWallHTCGnielinskiAnnularMaterial.C [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+{
+  const ADReal Pr = THM::Prandtl(_cp[_qp], _mu[_qp], _k[_qp]);
+
+  ADReal K;
+  if (_fluid_is_gas)
+  {
+    if (_T_wall[_qp] > _T[_qp] && !_provided_gas_heating_correction_exponent)
+      mooseError(
+          "If wall temperature ever exceeds the fluid temperature, even in iteration, and the "
+          "fluid is a gas, then the parameter 'gas_heating_correction_exponent' must be provided.");
```
Comment (licharlot) [inline modules/thermal_hydraulics/src/materials/ADWallHTCGnielinskiAnnularMaterial.C:96]: Isn't there a default value that could be used here?

File: modules/thermal_hydraulics/test/tests/materials/ad_wall_htc_gnielinski_annular/tests [hunk 3]
```diff
@@ -0,0 +1,33 @@
+[Tests]
+  design = 'ADWallHTCGnielinskiAnnularMaterial.md'
+  issues = '#26389'
+  [test]
+    requirement = "The system shall compute the heat transfer coefficient using the Gnielinski correlation for annular flows"
```
Comment (licharlot) [inline modules/thermal_hydraulics/test/tests/materials/ad_wall_htc_gnielinski_annular/tests:5]:
```suggestion
    requirement = "The system shall compute the heat transfer coefficient using the Gnielinski correlation for annular turbulent flows:"
```

File: modules/thermal_hydraulics/test/tests/materials/ad_wall_htc_gnielinski_annular/gold/ad_wall_htc_gnielinski_annular_out.csv [hunk 4]
```diff
@@ -0,0 +1,2 @@
+time,htc_wall
+0,5090.7825973763
```
Comment (licharlot) [inline modules/thermal_hydraulics/test/tests/materials/ad_wall_htc_gnielinski_annular/gold/ad_wall_htc_gnielinski_annular_out.csv:2]: This file is not used in the test specs.

File: modules/thermal_hydraulics/doc/content/source/materials/ADWallHTCGnielinskiAnnularMaterial.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+Lastly, the heat transfer coefficient is calculated as
+
+!equation
+h = \frac{\text{Nu} k}{D_h} \,.
+
+!syntax parameters /Materials/ADWallHeatTransferCoefficientGnielinskiMaterial
+
+!syntax inputs /Materials/ADWallHeatTransferCoefficientGnielinskiMaterial
+
+!syntax children /Materials/ADWallHeatTransferCoefficientGnielinskiMaterial
```
Comment (licharlot) [inline modules/thermal_hydraulics/doc/content/source/materials/ADWallHTCGnielinskiAnnularMaterial.md:58]:
```suggestion
!syntax parameters /Materials/ADWallHTCGnielinskiAnnularMaterial

!syntax inputs /Materials/ ADWallHTCGnielinskiAnnularMaterial

!syntax children /Materials/ ADWallHTCGnielinskiAnnularMaterial
```

File: modules/thermal_hydraulics/src/materials/ADWallHTCGnielinskiAnnularMaterial.C [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    K = std::pow(_T[_qp] / _T_wall[_qp], _n);
+  }
+  else
+  {
+    const ADReal cp_wall = _fp.cp_from_p_T(_p[_qp], _T[_qp]);
+    const ADReal mu_wall = _fp.mu_from_p_T(_p[_qp], _T[_qp]);
+    const ADReal k_wall = _fp.k_from_p_T(_p[_qp], _T[_qp]);
```
Comment (licharlot) [inline modules/thermal_hydraulics/src/materials/ADWallHTCGnielinskiAnnularMaterial.C:103]:
```suggestion
    const ADReal cp_wall = _fp.cp_from_p_T(_p[_qp], _T_wall[_qp]);
    const ADReal mu_wall = _fp.mu_from_p_T(_p[_qp], _T_wall[_qp]);
    const ADReal k_wall = _fp.k_from_p_T(_p[_qp], _T_wall[_qp]);
```

**Review by licharlot — CHANGES_REQUESTED** [review]

**Comment by licharlot** [comment]
Should we rename all the HTC coefficients using `HTC` instead of `HeatTransferCoefficient`?

File: modules/thermal_hydraulics/src/materials/ADWallHTCGnielinskiAnnularMaterial.C (hunk 2 above)
Comment (joshuahansel) [reply modules/thermal_hydraulics/src/materials/ADWallHTCGnielinskiAnnularMaterial.C:96]: There's not really a good default. I'd rather them be aware of the importance of this exponent than just use a bad default without knowing.

**Comment by joshuahansel** [comment]
> Should we rename all the HTC coefficients using `HTC` instead of `HeatTransferCoefficient`?

Probably so, I just don't want to do it here.

**Review by licharlot — APPROVED** [review]

**Pull request merged**